lib/src/utils/matrix_file.dart

```diff
@@ -233,12 +234,12 @@ class MatrixImageFile extends MatrixFile {
   /// you would likely want to use [NativeImplementations] and
   /// [Client.nativeImplementations] instead
   static MatrixImageFileResizedResponse? calcMetadataImplementation(
-      Uint8List bytes) {
-    final image = decodeImage(bytes);
+      MatrixImageFileCalcMetadataArguments arguments) {
+    final image = decodeNamedImage(arguments.fileName, arguments.bytes);
```

Comment:
sounds fancy but also error prone, we could try-catch this and if it fails use decodeImage instead

Reply:
well, the only case this will throw is when you send an image with a wrong extension. I'll do it anyways

Reply:
done

Reply:
I usually have lots of images with the wrong extension or no extension at all, does that properly handle that? Did you measure the performance impact of this? A smart decoder would just check the first bytes of each file, but I don't know if the dart one does that.